### PR TITLE
Re-enable coverage analysis

### PR DESCRIPTION
**Pull request by ylv-io**
Fixes https://github.com/OpenZeppelin/openzeppelin-contracts/issues/2189

Adds support for `solidity-coverage` using Test Environment library 🍾

### CODE REVIEW

File: scripts/coverage.sh [hunk 1]
```diff
@@ -1,25 +0,0 @@
...
-
-# The allFiredEvents file is created inside coverageEnv, but solidity-coverage
-# expects it to be at the top level. We create a symlink to fix this
-ln -s coverageEnv/allFiredEvents allFiredEvents
-
-OZ_TEST_ENV_COVERAGE=true npx solidity-coverage || log "Test run failed"
-
-if [ "$CI" = true ]; then
-  curl -s https://codecov.io/bash | bash -s -- -C "$CIRCLE_SHA1"
-fi
```
Comment (frangio) [inline scripts/coverage.sh:25]: We have to add this snippet in `coverage.js` to get coverage reports in PRs.

File: test/utils/Address.test.js [hunk 2]
```diff
@@ -138,6 +140,7 @@ describe('Address', function () {
       });
 
       it('reverts when the called function runs out of gas', async function () {
+        if (coverage) { return this.skip(); }
```
Comment (frangio) [inline test/utils/Address.test.js:143]: What's the reason we have to do this? Can you add a comment explaining that here?

Comment (nventuro) [reply test/utils/Address.test.js:143]: This test attempts to make the execution run out of gas, but the coverage EVM sets a ridiculously high gas limit. In practice, this ends up timeouting and causing all other tests to fail due to the hang in Ganache.

Comment (ylv-io) [reply test/utils/Address.test.js:143]: It is due to coverage setting block gas limit to `0xffffffffff` which cause `mockFunctionOutOfGas` function to crash Ganache and the subsequent tests before running out of gas.
```
  function mockFunctionOutOfGas() public payable {
        for (uint256 i = 0; ; ++i) {
            _array.push(i);
        }
    }
```

File: scripts/coverage.js [hunk 3]
```diff
@@ -1,5 +1,10 @@
 #!/usr/bin/env node
 
+const { execSync } = require('child_process');
 const { runCoverage } = require('@openzeppelin/test-environment');
 
-runCoverage(['mocks'], 'npm run compile', './node_modules/.bin/mocha --exit --timeout 10000 --recursive');
+runCoverage(['mocks'], 'npm run compile', './node_modules/.bin/mocha --exit --timeout 10000 --recursive'.split(' '));
+
+if (process.env.CI) {
+  execSync('curl -s https://codecov.io/bash | bash -s -- -C "$CIRCLE_SHA1"');
```
Comment (frangio) [inline scripts/coverage.js:9]: This is so we see the logs in CI.

```suggestion
  execSync('curl -s https://codecov.io/bash | bash -s -- -C "$CIRCLE_SHA1"', { stdio: 'inherit' });
```

**Comment by nventuro** [comment]
We're getting a strange error on the CI run:

```
TypeError: testCommand.split is not a function
 
    at runCoverage (/home/circleci/project/node_modules/@openzeppelin/test-environment/src/coverage.ts:56:37)
 
    at Object.<anonymous> (/home/circleci/project/scripts/coverage.js:6:1)
```

I see no call to `split` [on line 56](https://github.com/OpenZeppelin/openzeppelin-test-environment/blob/feature/solidity-coverage/src/coverage.ts#L56). I'm not sure what's up with that.

**Comment by ylv-io** [comment]
It is because it can't realize that `test-env` has changed. Once @frangio approves `test-env` PR we'll have a proper dependency and it will work.

**Pull request merged**